### PR TITLE
Bugfix 0669648 normalize user input for configset

### PR DESCRIPTION
**Pull request by JaydenLiang**
fixed #9 (regarding quotation marks in psksecret)
fixed #mantis0669648 (regarding handling replaceable portion of configset)

### CODE REVIEW

File: fortigate-autoscale/fortigate-bootstrap-config-strategy.ts [hunk 1]
```diff
@@ -272,23 +272,40 @@ export class FortiGateBootstrapConfigStrategy implements BootstrapConfigurationS
             config = this.processConfigV2(config, sourceData);
         }
 
-        const psksecret = this.settings.get(FortiGateAutoscaleSetting.FortiGatePskSecret).value;
+        const psksecret = this.normalizeFOSCmdInput(
```
Comment (jamie-pate) [inline fortigate-autoscale/fortigate-bootstrap-config-strategy.ts:275]: maybe you could just make a copy of FortiGateAutoscaleSettings where everything is normalized?

`normalizedSettings = Object.entries(FortiGateAutoscaleSettings).reduce((r, [key, value]) => { r[key] = this.normalizeFOSCmdInput(value); return r}, {})`

File: fortigate-autoscale/fortigate-bootstrap-config-strategy.ts [hunk 2]
```diff
@@ -336,6 +353,16 @@ export class FortiGateBootstrapConfigStrategy implements BootstrapConfigurationS
...
     }
+
+    /**
+     * To normalize a string in order for a safe use as the input value for the FOS commands.
+     *
+     * @param {string} input the input string to normalize
+     * @returns {string} the normalized input string
+     */
+    protected normalizeFOSCmdInput(input: string): string {
+        return (input && input.replace(/\\/g, '\\\\').replace(/"/g, '\\"')) || null;
```
Comment (jamie-pate) [inline fortigate-autoscale/fortigate-bootstrap-config-strategy.ts:364]: there's probably other special characters?

I usually do something like
```
input.replace(/\\"'/g, m => `\\${m}`)
```

https://help.fortinet.com/fweb/4-3-5/cli/wwhelp/wwhimpl/common/html/wwhelp.htm?context=FortiWeb_Online_CLI_Reference&file=CLI_basics_3_15.html

**Review by jamie-pate — CHANGES_REQUESTED** [review]

File: fortigate-autoscale/fortigate-bootstrap-config-strategy.ts (hunk 2 above)
Comment (JaydenLiang) [reply fortigate-autoscale/fortigate-bootstrap-config-strategy.ts:364]: this link is very helpful. I'll update the code.

File: fortigate-autoscale/fortigate-bootstrap-config-strategy.ts (hunk 1 above)
Comment (JaydenLiang) [reply fortigate-autoscale/fortigate-bootstrap-config-strategy.ts:275]: I'll make an improvement for it.

**Review by jamie-pate — APPROVED** [review]

**Pull request merged**